runhouse/resources/secrets/secret.py

```diff
@@ -182,28 +179,45 @@ def extract_provider_secrets(cls, names: List[str] = None) -> Dict[str, "Secret"
         return secrets
 
     # TODO: refactor this code to reuse rns_client save_config code instead of rewriting
-    def save(self, values: bool = True, headers: str = rns_client.request_headers):
+    def save(self, save_values: bool = True, headers: Optional[str] = None):
         """
-        Save the secret config, into Vault if the user is logged in,
+        Save the secret config to Den. Save the secret values into Vault if the user is logged in,
         or to local if not or if the resource is a local resource.
         """
         config = self.config_for_rns
         config["name"] = self.rns_address
-        if values:
-            config["values"] = self.values
+        headers = headers or rns_client.request_headers
 
+        # Save metadata to Den
         if self.rns_address.startswith("/"):
-            logger.info(f"Saving config for {self.name} to Vault")
+            logger.info(f"Saving config for {self.name} to Den")
             payload = rns_client.resource_request_payload(config)
-            resp = requests.put(
-                f"{rns_client.api_server_url}/{self.USER_ENDPOINT}/{self.name}",
+            resp = requests.post(
+                f"{rns_client.api_server_url}/resource",
                 data=json.dumps(payload),
                 headers=headers,
             )
+            # If resource already saved in Den will return a 422
+            if resp.status_code not in [200, 422]:
```

Comment:
I think this is fine, means we just create or update the resource in Den. Not sure if there's any edge cases we might need to handle for re-saving existing secrets given the decoupling of config and secrets? (ex/ same rns config, diff secret values)